Backport 'Add titles in initiatives' public pages' to v0.26

andreslucena opened the pull request:
#### :tophat: What? Why?

Backport #11809 to v0.26

:hearts: Thank you!

alecslupu reviewed (commented):
@andreslucena This PR is totally different than the PR meant to backport. Are you sure we do not introduce a change, featured as backport?

andreslucena commented:
> @andreslucena This PR is totally different than the PR meant to backport. 

I think we will start to have much more differences in the backports, as specially the changes that touch the HTML will necessarily change a lot, just for the diff between versions (redesign and old design).

> Are you sure we do not introduce a change, featured as backport?

To be honest while I was doing this one I wasn't 100% sure on how to approach it, as there aren't clear titles (`h1`) in the markup. 

For a better understanding, this is how it looks in v0.28.0.dev with the redesign (in nightly.decidim.org): 
 
![Screenshot of the "Select an initiative" page in v0.28](https://github.com/decidim/decidim/assets/717367/56c50f99-fdff-4fcd-b4aa-3b6a787a7e5c)

And this is how it looks like in v0.26/v0.27 (in try.decidim.org): 

![Screenshot of the "Select an initiative" page in v0.27](https://github.com/decidim/decidim/assets/717367/a39b741d-1094-41d3-875a-fa8c4a76585a)

So, as it isn't that clear what the HTML document title should be as there isn't any h1 in the current markup, I buy the argument that this isn't clean and easy to adapt and backport, so it's better to not do so. So, let's close this PR. 

What I don't agree conceptually is with the concept of `type: change` generally. IMO every PR merged is a change and every introduction in the changelog is obviously a change too (hence the name!! CHANGE-log). But yeah, let's say that in these changes the blur line between `type: bug`, `type: feature` and `type: change` get blurrier. 

According to keepachangelog.com, what we should use is:

    Added for new features.
    Changed for changes in existing functionality. 

But yeah, I don't know how to differentiate what's a new feature or a change in an existing one. Maybe it's not that important as long as we're sure and happy on which things we backport and to raise our hands when we have doubts (as you did in this PR). 

Closing this PR as we will not merge it.
